chore: update loki to 3.1.1

chance-coleman opened the pull request:
## Description
Updates for bringing in loki renovate updates

## Related Issue

Fixes #308 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

chance-coleman committed:
fix: initial commit for loki updates

chance-coleman committed:
fix: lint error

chance-coleman committed:
fix: update registry1 image

chance-coleman committed:
fix: pod anti affinity

chance-coleman committed:
fix: remove deploymentMode value

chance-coleman committed:
fix: revert removal of deploymentMode

mjnagel committed:
Merge branch 'main' into renovate-loki-update

mjnagel committed:
Merge branch 'main' into renovate-loki-update

mjnagel committed:
Merge branch 'main' into renovate-loki-update

mjnagel commented on src/loki/values/values.yaml:
```diff
@@ -67,6 +76,7 @@ test:
   enabled: false
 
 # Configuration for the single binary node(s)
+deploymentMode: SingleBinary
```
I think this is a change from the previous? In core before we were defaulting to `SimpleScalable` (although it was done a different way in the past). We likely want to default to scalable still...

mjnagel commented on src/loki/values/values.yaml:
```diff
@@ -25,6 +25,15 @@ loki:
     replication_factor: 1
+  schemaConfig:
+    configs:
+      - from: 2024-04-01
```
I think this should be set to a date in the future based on https://grafana.com/docs/loki/latest/operations/storage/schema/#changing-the-schema, otherwise it will result in logs from 4/1 -> today being unreadable since they are actually v12 schema? I think we could set it to one month+ out explicitly, OR we could look at doing this via zarf var/pepr to mutate in a date in the future.

I'm not certain but we may also need a v12 schema entry here to handle older logs (using a `from` date of something last year to safely handle all possible logs from the first uds-core release)?

mjnagel reviewed (commented):
Haven't fully deployed yet but wanted to get these two comments out there.

chance-coleman committed:
fix: pr comments

mjnagel reviewed (commented):
We probably also want to switch over to `Secret` for this value - https://github.com/grafana/loki/blob/main/production/helm/loki/values.yaml#L112

chance-coleman committed:
fix: change to secret and fix date to resolve for now

chance-coleman committed:
fix: break out test apps for less ci cpu usage

chance-coleman committed:
fix: lint

chance-coleman committed:
fix: initial commit for loki operator

chance-coleman committed:
Merge branch 'main' into renovate-loki-update

chance-coleman committed:
fix: lint

chance-coleman committed:
fix: duplicate prometheus

chance-coleman committed:
fix: update to use child logger

chance-coleman committed:
fix: lint again

chance-coleman committed:
fix: authservice test app stuff

chance-coleman committed:
Merge branch 'main' into renovate-loki-update

chance-coleman committed:
fix: app-tenant ref

chance-coleman committed:
fix: nginx image failure

chance-coleman committed:
fix: unicorn nginx

chance-coleman committed:
fix: add unicorn memcached image

chance-coleman committed:
fix:lint

chance-coleman committed:
fix: lint again...

chance-coleman committed:
fix: size of memcached and memcache image

chance-coleman committed:
fix: disable resultsCache

chance-coleman committed:
Merge branch 'main' into renovate-loki-update

mjnagel commented on src/pepr/loki/index.ts:
```diff
@@ -0,0 +1,107 @@
+
+            // Update the secret with the new config.yaml content
+            secret.Raw.data["config.yaml"] = yaml.dump(lokiConfig);
+            log.info(secret.Raw.data["config.yaml"], `Secret config.yaml updated successfully.`);
```
I think there might be some missing logic here - but want to summarize my understanding of the cases and how we should be handling them:
- install/upgrade: the date (as set in this PR) would be a previous date (2024-06-04) and we would want to generate a future date, 2 days in advance as discussed to be safe with the way Loki handles this
- future upgrade: the date coming into the webhook would be our previous date (2024-06-04), we would want to set this to the date we already generated (probably could query the cluster for this?)

The key here is that we want to generate a date in the future once, but once we generate it once always use that date for future mutations (since the baseline applied from the chart will always be an older date). I think in the current state we would end up mutating this date any time it is in the past?

chance-coleman replied:
Yeah that is a good point that i totally misunderstood. I've updated so the date is only generated once and stored as an annotation in the secret for future reference.

chance-coleman committed:
Merge branch 'main' into renovate-loki-update

chance-coleman committed:
fix: date logic, create readme

chance-coleman committed:
fix: lint

chance-coleman committed:
Merge branch 'main' into renovate-loki-update

chance-coleman committed:
Merge branch 'main' into renovate-loki-update

chance-coleman committed:
Merge branch 'main' into renovate-loki-update

chance-coleman committed:
fix: ci test app deployment

chance-coleman committed:
Merge branch 'main' into renovate-loki-update

rjferguson21 committed:
chore: update loki (#734)

## Description

...

## Related Issue

Fixes #
<!-- or -->
Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

rjferguson21 committed:
Merge branch 'main' into renovate-loki-update

rjferguson21 committed:
remove remaining pepr changes

rjferguson21 committed:
roll back test apps changes

rjferguson21 committed:
remove test apps

rjferguson21 committed:
Merge branch 'main' into renovate-loki-update

rjferguson21 committed:
Merge remote-tracking branch 'origin/main' into renovate-loki-update

rjferguson21 committed:
Merge branch 'main' into renovate-loki-update

rjferguson21 committed:
fix loki metrics

rjferguson21 committed:
remove remoteCidr for test app

chance-coleman commented on src/test/app-tenant.yaml:
```diff
@@ -26,7 +26,6 @@ spec:
       - service: test-tenant-app-cidr
         selector:
           app: test-tenant-app
-        remoteCidr: "192.168.0.0/16"
```
This [was added ](https://github.com/defenseunicorns/uds-core/pull/719/files#diff-c94db7cc6e40995709784aae6c1a7f05ae2ed175ca1c0eb2553de353ddbd5d5fL25-L26)for testing that the remoteCidr generation wouldn't break anything. If you're going to remove this then the whole netpol should be removed since that was its only usecase.

chance-coleman committed:
chore: remove broken netpol
